state/pgstatestorage.go

```diff
@@ -421,19 +421,43 @@ func (p *PostgresStorage) GetLastNBatches(ctx context.Context, numBatches uint,
 }
 
 // GetLastNBatchesByL2BlockNumber returns the last numBatches batches along with the l2 block state root by l2BlockNumber
-func (p *PostgresStorage) GetLastNBatchesByL2BlockNumber(ctx context.Context, l2BlockNumber uint64, numBatches uint, dbTx pgx.Tx) ([]*Batch, common.Hash, error) {
+// if the l2BlockNumber parameter is nil, it means we want to get the most recent last N batches
```

Comment:
IMO since we already have a function that `get the most recent last N batches` I'd keep `GetLastNBatchesByL2BlockNumber` "clean" (without the particular behavior when `l2BlockNumber == nil`) and let the consumer choose which function has to use. Although this is just an opinion, here is my rationale:

- More predictable: what the function does is closer to its name
- Easier: 2 easy SQL queries vs 1 complex query. IMO it's easier to debug and maintain 2 easy